firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -1178,4 +1178,171 @@ public void testOrQueriesWithArrayMembership() {
         "doc4",
         "doc6");
   }
+
+  @Ignore
+  @Test
+  public void testMultipleInOps() {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "doc1", map("a", 1, "b", 0),
+            "doc2", map("b", 1),
+            "doc3", map("a", 3, "b", 2),
+            "doc4", map("a", 1, "b", 3),
+            "doc5", map("a", 1),
+            "doc6", map("a", 2));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+
+    // Two IN operations on different fields with disjunction.
+    Query query1 =
+        collection
+            .where(Filter.or(Filter.inArray("a", asList(2, 3)), Filter.inArray("b", asList(0, 2))))
+            .orderBy("a");
+    checkOnlineAndOfflineResultsMatch(query1, "doc1", "doc6", "doc3");
+
+    // Two IN operations on different fields with conjunction.
+    Query query2 =
+        collection
+            .where(Filter.and(Filter.inArray("a", asList(2, 3)), Filter.inArray("b", asList(0, 2))))
+            .orderBy("a");
+    checkOnlineAndOfflineResultsMatch(query2, "doc3");
+
+    // Two IN operations on the same field.
+    // a IN [1,2,3] && a IN [0,1,4] should result in "a==1".
+    Query query3 =
+        collection.where(
+            Filter.and(Filter.inArray("a", asList(1, 2, 3)), Filter.inArray("a", asList(0, 1, 4))));
+    checkOnlineAndOfflineResultsMatch(query3, "doc1", "doc4", "doc5");
+
+    // a IN [2,3] && a IN [0,1,4] is never true and so the result should be an empty set.
+    Query query4 =
+        collection.where(
+            Filter.and(Filter.inArray("a", asList(2, 3)), Filter.inArray("a", asList(0, 1, 4))));
+    checkOnlineAndOfflineResultsMatch(query4);
+
+    // a IN [0,3] || a IN [0,2] should union them (similar to: a IN [0,2,3]).
+    Query query5 =
+        collection.where(
+            Filter.or(Filter.inArray("a", asList(0, 3)), Filter.inArray("a", asList(0, 2))));
+    checkOnlineAndOfflineResultsMatch(query5, "doc3", "doc6");
+
+    // Nested composite filter on the same field.
+    Query query6 =
+        collection.where(
+            Filter.and(
+                Filter.inArray("a", asList(1, 3)),
+                Filter.or(
+                    Filter.inArray("a", asList(0, 2)),
+                    Filter.and(
+                        Filter.greaterThanOrEqualTo("b", 1), Filter.inArray("a", asList(1, 3))))));
+    checkOnlineAndOfflineResultsMatch(query6, "doc3", "doc4");
+
+    // Nested composite filter on different fields.
```

Comment:
This seems very similar to "Nested composite filter on the same field." But, it's not a big deal.

Reply:
the inner-most IN filter is on field `a` in both cases. The parent IN Filters in the former is also on `a`, but on `b` in the latter. Both should be valid queries.